src/BackendRequest.h

```diff
@@ -40,15 +40,17 @@ namespace NextPVR
       return request;
     }
     int DoRequest(const char* resource, std::string& response);
-    tinyxml2::XMLError DoMethodRequest(const char* resource, tinyxml2::XMLDocument& doc);
+    tinyxml2::XMLError DoMethodRequest(const char* resource, tinyxml2::XMLDocument& doc, bool compresssed = true);
     int FileCopy(const char* resource, std::string fileName);
-    void setSID(char* newsid) { strcpy(m_sid, newsid); };
-    void clearSid() { std::fill(std::begin(m_sid), std::end(m_sid), 0); m_sidUpdate = 0; };
+    tinyxml2::XMLError  GetLastUpdate(const char* resource, time_t& last_update);
     bool PingBackend();
     bool OneTimeSetup();
-    const char* getSID() { return m_sid; };
+    const char* getSID() { return m_sid.c_str(); };
     std::vector<std::vector<std::string>> Discovery();
-    bool isSidActive() { return time(nullptr) < m_sidUpdate + 900; };
+
+    void setSID(std::string newsid) { m_sid = newsid; };
+    void clearSid() { m_sid.clear(); m_sidUpdate = 0; };
+    bool isSidActive() { return !m_sid.empty() && time(nullptr) < m_sidUpdate + 900; };
```

Comment:
`IsSidActive`

Reply:
Ok I will adjust them all to be consistent.

Reply:
Next release SID functions updated